### ISSUE TITLE
new masking techniques

### ISSUE THREAD

**Issue by vmazashvili** [vmazashvili/Neural-Networks#6]
Ideas:

1. Top-k Saliency Masking
2. Channel-wise Masking
3. Guided Backpropagation Masking
4. Random Erasing
5. CutOut

**Comment by vmazashvili** [comment]
modify the code in a way that I can generate the datasets and save them in masked_train_dataloaders with the masking function and parameter identifier. for example, in fine tuning loop, ill be able to save each finetuned model checkpoint for each dataset 
torch.save(model.state_dict(), f'/content/data/MNIST/CNN_checkpoint_finetuned_{method}_{parameter}.pth')

**Comment by vmazashvili** [comment]
added:
Top-k Saliency Masking
Channel-wise Masking

**Comment by vmazashvili** [comment]
added:
Top-k Saliency Masking
Channel-wise Masking
